api/valor_api/crud/_update.py

```diff
@@ -21,3 +29,8 @@ def finalize(*, db: Session, dataset_name: str, model_name: str | None = None):
             name=dataset_name,
             status=enums.TableStatus.FINALIZED,
         )
+
+    if task_handler:
+        task_handler.add_task(vacuum_analyze)
```

Comment:
what's the rationale for only doing this on finalization? would there be some benefit to doing it after x evaluation runs?

Reply:
The rationale for adding vacuum analyze to only the dataset/model finalization is based on the magnitude of table restructuring. The data added in dataset/model creation is used directly in computations so understanding the statistics of those tables (and related ones) is important to ensuring a consistent evaluation time.

Evaluations themselves do not produce data other than metrics. These metrics don't affect computation time and therefore there is little to no benefit to adding the same operation to them since we already implement indexing on these tables.

Reply:
makes sense! thanks for the write-up